### PR TITLE
Support commenting using `giscus`

### LINKED ISSUE #536: Add support for `giscuss` a `utterances` alternative that uses GitHub discussion

### PR DESCRIPTION
**Pull request by Rihoj**
### Issue
Fixes #536 

### Description
Adds [Giscus](https://giscus.app/) support
I do think a guide post should be added, but I wanted to make sure you like how this is done before I start writing out a post. (Unless someone else wants to write it)

### Test Evidence
image

### CODE REVIEW

File: layouts/partials/giscus.html [hunk 1]
```diff
@@ -0,0 +1,21 @@
+{{ $repo := site.Params.features.comment.giscus.repo }}
+{{ $category := site.Params.features.comment.giscus.category | default "Announcements" }}
```
Comment (hossainemruz) [inline layouts/partials/giscus.html:2]: Let's use `General`

File: layouts/partials/giscus.html [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+{{ $theme := site.Params.features.comment.giscus.theme | default "dark" }}
```
Comment (hossainemruz) [inline layouts/partials/giscus.html:3]: Please use `light` theme by default.

**Review by hossainemruz — CHANGES_REQUESTED** [review]
Thank you @Rihoj. Looks good to me. Just a little change. Can you please acknowledge the comments?

**Review by hossainemruz — APPROVED** [review]

**Comment by hossainemruz** [comment]
Thank you @Rihoj . Looks good to me. Would you like to have a PR in guide repo? I can merge this PR.

**Comment by Rihoj** [comment]
@hossainemruz I am currently working on the guide PR. Since the structure was a single page for Analytics and Comments, I am working on detailing the whole page out, so it will be a little longer as I am not always elegant with my words.

**Comment by hossainemruz** [comment]
Yes. It will be good idea to separate the analytics and comment post.

**Comment by roumail** [comment]
I'm unable to see comments on my website following these instructions for giscus. Any ideas what could have gone wrong?

**Comment by roumail** [comment]
Switched to utteranc and it still doesn't work...

**Pull request merged**